### PR TITLE
Rename 'rocks' package to 'database' to clarify responsibility

### PR DESCRIPTION
**Pull request by lolski**
## What is the goal of this PR?

We make a clearer architecture for the 'rocks' package, reflecting its responsibility which is managing databases.  

This improvement contains two major changes: 

1. Merged `RocksTypeDB` into `RocksDatabaseManager`. The former was a very thin class that did nothing but instantiate RocksDB and `RocksDatabaseManager`.
2. Updated the package and the class names:
    - Renamed the package 'rocks' to `database`
    - Prefixed top-level classes with the word `Core`
    - Prefixed low-level classes that deals with RocksDB with the word `Rocks`

## What are the changes implemented in this PR?

- Merged `RocksTypeDB` into `RocksDatabaseManager`
  - Updated `TypeDB` interface accordingly
  - Updated `Factory` interface and implementation accordingly
  - Updated `TypeDBService` accordingly

- Prefixed the top-level classes with the word `Core` :
  - `RocksDatabaseManager` -> `CoreDatabaseManager`
  - `RocksDatabase` -> `CoreDatabase`
  - `RocksSession` -> `CoreSession`
  - `RocksTransaction` -> `CoreTransaction`
- Prefixed the low-level classes that mostly deals with RocksDB, with the word `Rocks`:
  - `Configuration` -> `RocksConfiguration`
  - `RocksPartitionManager`
  - `RocksStorage`
  - `RocksIterator`
  - `RocksProperties`

- Updated `TypeDBService` and tests accordingly to reflect the new merged class and renames

### CODE REVIEW

File: database/CoreDatabase.java [hunk 1]
```diff
@@ -84,33 +84,33 @@
...
     protected RocksPartitionManager.Data rocksDataPartitionMgr;
-    protected RocksSession.Data statisticsBackgroundCounterSession;
+    protected CoreSession.Data statisticsBackgroundCounterSession;
     protected StatisticsBackgroundCounter statisticsBackgroundCounter;
     protected ScheduledExecutorService scheduledPropertiesLogger;
     private Cache cache;
 
-    protected RocksDatabase(RocksTypeDB typedb, String name, Factory.Session sessionFactory) {
-        this.typedb = typedb;
+    protected CoreDatabase(CoreDatabaseManager databaseManager, String name, Factory.Session sessionFactory) {
```
Comment (flyingsilverfin) [inline database/CoreDatabase.java:112]: naming `databaseMgr`

File: database/CoreDatabaseManager.java [hunk 2]
```diff
@@ -32,29 +39,57 @@
...
+        RocksDB.loadLibrary();
+        Loader.loadNativeLibraries();
+        ErrorMessage.loadConstants();
+    }
 
     protected static final String RESERVED_NAME_PREFIX = "_";
 
-    protected final RocksTypeDB typedb;
-    protected final ConcurrentMap<String, RocksDatabase> databases;
+    private final Options.Database typeDBOptions;
```
Comment (flyingsilverfin) [inline database/CoreDatabaseManager.java:56]: `databaseOptions`?

File: database/RocksStorage.java [hunk 3]
```diff
@@ -57,8 +57,6 @@
 import static com.vaticle.typedb.core.common.exception.ErrorMessage.Internal.RESOURCE_CLOSED;
 import static com.vaticle.typedb.core.common.exception.ErrorMessage.Transaction.TRANSACTION_DATA_READ_VIOLATION;
 import static com.vaticle.typedb.core.common.exception.ErrorMessage.Transaction.TRANSACTION_SCHEMA_READ_VIOLATION;
-import static com.vaticle.typedb.core.graph.common.Encoding.ENCODING_VERSION;
-import static com.vaticle.typedb.core.graph.common.Encoding.System.ENCODING_VERSION_KEY;
```
Comment (flyingsilverfin) [inline database/RocksStorage.java:61]: ah shoot looks like I forgot to clean up some imports!! Good catch!

File: migrator/MigratorService.java [hunk 4]
```diff
@@ -32,17 +32,17 @@
 public class MigratorService extends MigratorGrpc.MigratorImplBase {
     private static final Logger LOG = LoggerFactory.getLogger(MigratorService.class);
 
-    private final TypeDB typedb;
+    private final TypeDB.DatabaseManager databaseManager;
     private final String version;
 
-    public MigratorService(TypeDB typedb, String version) {
-        this.typedb = typedb;
+    public MigratorService(TypeDB.DatabaseManager databaseManager, String version) {
```
Comment (flyingsilverfin) [inline migrator/MigratorService.java:38]: (more naming)

File: server/conf/typedb.yml [hunk 5]
```diff
@@ -45,7 +45,7 @@ log:
       level: info
       output: [ stdout, file ]
     storage:
-      filter: com.vaticle.typedb.core.rocks
+      filter: com.vaticle.typedb.core.database
       level: info # on 'debug' the server will periodically log storage properties
```
Comment (flyingsilverfin) [inline server/conf/typedb.yml:49]: maybe this comment should be updated to read `... log database storage properties`?

File: server/conf/typedb.yml [hunk 6]
```diff
@@ -45,8 +45,8 @@ log:
       level: info
       output: [ stdout, file ]
     storage:
-      filter: com.vaticle.typedb.core.rocks
-      level: info # on 'debug' the server will periodically log storage properties
+      filter: com.vaticle.typedb.core.database
```
Comment (lolski) [inline server/conf/typedb.yml:48]: Update the log filter according to the new package name.

File: server/TypeDBServer.java [hunk 7]
```diff
@@ -68,14 +68,14 @@
     private static final Logger LOG = LoggerFactory.getLogger(TypeDBServer.class);
 
     protected final Factory factory;
-    protected final TypeDB typedb;
+    protected final TypeDB.DatabaseManager databaseMgr;
```
Comment (lolski) [inline server/TypeDBServer.java:71]: Now that `RocksTypeDB` has been deleted, `databaseManager` is owned by `TypeDBServer` directly.

File: rocks/RocksTypeDB.java [hunk 8]
```diff
@@ -1,114 +0,0 @@
-/*
```
Comment (lolski) [inline rocks/RocksTypeDB.java:1]: This class has been deleted, and have the main functionalities moved to `CoreDatabaseManager`.

File: migrator/MigratorService.java [hunk 9]
```diff
@@ -32,17 +32,17 @@
 public class MigratorService extends MigratorGrpc.MigratorImplBase {
     private static final Logger LOG = LoggerFactory.getLogger(MigratorService.class);
 
-    private final TypeDB typedb;
+    private final TypeDB.DatabaseManager databaseMgr;
```
Comment (lolski) [inline migrator/MigratorService.java:35]: Now that `RocksTypeDB` has been deleted, the reference is updated to point to `databaseManager`.

File: migrator/data/DataExporter.java [hunk 10]
```diff
@@ -49,7 +49,7 @@
 public class DataExporter {
     private static final Logger LOG = LoggerFactory.getLogger(DataExporter.class);
 
-    private final TypeDB typedb;
+    private final TypeDB.DatabaseManager databaseMgr;
```
Comment (lolski) [inline migrator/data/DataExporter.java:52]: Now that `RocksTypeDB` has been deleted, the reference is updated to point to `databaseManager`.

File: database/CoreDatabaseManager.java [hunk 11]
```diff
@@ -32,29 +39,57 @@
...
+    protected CoreDatabaseManager(Options.Database databaseOptions, Factory.Database databaseFactory) {
+        if (!Executors.isInitialised()) Executors.initialise(MAX_THREADS);
+        this.databaseOptions = databaseOptions;
         this.databaseFactory = databaseFactory;
         databases = new ConcurrentHashMap<>();
         isOpen = new AtomicBoolean(true);
+        loadAll();
+    }
+
+    @Override
```
Comment (lolski) [inline database/CoreDatabaseManager.java:82]: Moved from `RocksTypeDB`.

**Pull request merged**